chef/automate
# Jamie/1841 search filter global projects

### :nut_and_bolt: Description: What code changed, and why?
This branch adds a filtering feature to our global projects dropdown menu.  This allows users to sort through their open projects by name and check/uncheck which projects they would like to view.

### :chains: Related Resources
https://github.com/chef/automate/issues/1841

### :+1: Definition of Done
User can filter through their projects inside the dropdown, select, deselect, and apply selection to view their chosen projects limited by their filter choice.

Please note: in the interest of deadlines, the Create Project and Edit Project buttons have been moved to a later release.

### :athletic_shoe: How to Build and Test the Change
To Build
1. build components/automate-ui-devproxy && start_all_services
2. make serve

To Test
Its easier to visualize if you navigate to https://a2-dev.test/settings/projects
if you do not have several projects, please create at least 3
Click the projects dropdown button in the top right of the screen
in the dropdown, you should see an input box, begin to type the name of a project
you should see the list begin to filter closer to your project name as you type

select one project by clicking the checkbox next to its name
you should see a blue check mark, the project count raise by 1
the name of the project be the new title on the dropdown.

select two projects
you should see the project count = 2
the new title should read "multiple projects"

click the "apply" button
you should see page refresh
only your chosen projects should now be viewable

click the "Clear selection" button
you should see page refresh
all projects are now visible again

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable

![Screen Shot 2020-01-21 at 2 11 01 PM](https://user-images.githubusercontent.com/16737484/72847720-f3e65000-3c57-11ea-8e0e-eb999435a96a.png)
![Screen Shot 2020-01-17 at 1 56 47 PM](https://user-images.githubusercontent.com/16737484/72847721-f3e65000-3c57-11ea-82d4-f3fff84206be.png)
![Screen Shot 2020-01-21 at 2 10 53 PM](https://user-images.githubusercontent.com/16737484/72847722-f3e65000-3c57-11ea-9355-16251298150f.png)
